proposals/0035-linalg-matrix.md

```diff
@@ -859,4 +923,216 @@ Stores a row-vector to a matrix. Out of bounds writes no-op.
   store operations that operate on descriptors or should we assume
   DXILMatrixComponentType?
 
+## Appendix 2: HLSL Header
+
+[Compiler Explorer](https://godbolt.org/z/zTbfvPPqP)
+> Note: this mostly works with Clang, but has some issues to work out still.
+
+```cpp
+namespace hlsl {
+
+template <typename T> struct is_arithmetic {
+  static const bool value = false;
+};
+
+#define __ARITHMETIC_TYPE(type)                                                \
+  template <> struct is_arithmetic<type> {                                     \
+    static const bool value = true;                                            \
+  };
+
+#if __HLSL_ENABLE_16_BIT
+__ARITHMETIC_TYPE(uint16_t)
+__ARITHMETIC_TYPE(int16_t)
+#endif
+__ARITHMETIC_TYPE(uint)
+__ARITHMETIC_TYPE(int)
+__ARITHMETIC_TYPE(uint64_t)
+__ARITHMETIC_TYPE(int64_t)
+__ARITHMETIC_TYPE(half)
+__ARITHMETIC_TYPE(float)
+__ARITHMETIC_TYPE(double)
+
+template <bool B, typename T> struct enable_if {};
+
+template <typename T> struct enable_if<true, T> {
+  using type = T;
+};
+
+} // namespace hlsl
+
+namespace dx {
+
+namespace linalg {
+
+enum class MatrixComponentType {
+  Invalid = 0,
+  I1 = 1,
+  I16 = 2,
+  U16 = 3,
+  I32 = 4,
+  U32 = 5,
+  I64 = 6,
+  U64 = 7,
+  F16 = 8,
+  F32 = 9,
+  F64 = 10,
+  SNormF16 = 11,
+  UNormF16 = 12,
+  SNormF32 = 13,
+  UNormF32 = 14,
+  SNormF64 = 15,
+  UNormF64 = 16,
+  PackedS8x32 = 17,
+  PackedU8x32 = 18,
+};
+
+namespace __detail {
+template <MatrixComponentType T> struct ComponentTypeTraits {
+  using Type = uint;
+  static const bool IsNativeScalar = false;
+};

```

Comment:
Why isn't there a struct for, say, `int`? I don't see a situation where `using Type = int` would exist. 
It also isn't represented in any of the examples below, so I'm wondering if this is an example list or if it's supposed to be complete?
Or maybe it's supposed to be `using Type = T`?

Reply:
The specialization at the top just allows treating any of the packed types as `uint`. There is a mapping of `I32` to `int32_t`. We don't need to have separate handling for `int` since they're the same type.